packages/devtools/lib/src/memory/memory.dart

```diff
@@ -362,38 +682,357 @@ class MemoryScreen extends Screen with SetStateMixin {
+      return;
+    }
+
+    // Hover in the cell.
+    if (hover.data != _currentHoverSummary?.data) {
+      // Selecting a different instance that what's current.
+      _closeHover(hover);
+    }
+
+    // Entering Hover again?
+    if (hoverPopup.element.children.isNotEmpty) return;
+
+    final CoreElement ulElem = ul();
+    final InboundReferences refs =
+        await getInboundReferences(hover.data.objectRef, 1000);
+
+    if (refs == null) {
+      framework.toast("Instance ${hover.data.objectRef} - Sentinel/Expired.",
+          title: 'Warning');
+      return;
+    }
+
+    ulElem.add(li(c: 'allocation-li-title')
+      ..add([
+        span(text: 'Allocated', c: 'allocated-by-class-title'),
+        span(text: 'Referenced', c: 'ref-by-title')
+      ]));
+
+    for (InboundReference element in refs.elements) {
+      // Could be a reference to an evaluate so this isn't known.
+
+      // Looks like an object created from an evaluate, ignore it.
+      if (element.parentField == null && element.json == null) continue;
+
+      // TODO(terry): Verify looks like internal class (maybe to C code).
+      if (element.parentField.owner != null &&
+          element.parentField.owner.name.contains('&')) continue;
+
+      String referenceName;
+      String owningAllocator; // Class or library that allocated.
+      bool owningAllocatorIsAbstract;
+
+      switch (element.parentField.runtimeType.toString()) {
+        case 'ClassRef':
+          final ClassRef classRef = element.classRef;
+          owningAllocator = classRef.name;
+          // TODO(terry): Quick way to detect if class is probably abstract-
+          // TODO(terry): Does it exist in the class list table?
+          owningAllocatorIsAbstract = _searchClass(owningAllocator) == null;
+          break;
+        case 'FieldRef':
+          final FieldRef fieldRef = element.fieldRef;
+          referenceName = fieldRef.name;
+          switch (fieldRef.owner.runtimeType.toString()) {
+            case 'ClassRef':
+              final ClassRef classRef = ClassRef.parse(fieldRef.owner.json);
+              owningAllocator = classRef.name;
+              // TODO(terry): Quick way to detect if class is probably abstract-
+              // TODO(terry): Does it exist in the class list table?
+              owningAllocatorIsAbstract = _searchClass(owningAllocator) == null;
+              break;
+            case 'Library':
+            case 'LibraryRef':
+              final Library library = Library.parse(fieldRef.owner.json);
+              owningAllocator =
+                  'Library ${library != null ? library.name : ""}';
+              break;
+          }
+          break;
+        case 'FuncRef':
+          print('Error: Unhandled ${element.parentField.runtimeType}');
+          // TODO(terry): TBD
+          // final FuncRef funcRef = element.funcRef;
+          break;
+        case 'Instance':
+          print('Error: Unhandled ${element.parentField.runtimeType}');
```

Comment:
eliminate all print statement errors and log to analytics instead?

Reply:
Done.